src/index.cpp

```diff
@@ -327,6 +328,60 @@ namespace diskann {
     }
   }
 
+  template<typename T, typename TagT>
+  void Index<T, TagT>::select_starting_points(
+      Parameters &parameters,
+      std::mt19937 &gen,
+      std::set<unsigned> &unique_start_points) {
+
+    auto strategy = parameters.Get<std::string>(
+        Constants::selection_strategy_of_starting_points,
+        Constants::random);
+    auto num_starting_points =
+        parameters.Get<unsigned>(Constants::num_starting_points, 0);
+    if (num_starting_points == 0) {
+      return;
+    }
+
+    if (_nd < num_starting_points + 1) {
```

Comment:
In streaming settings, the index starts with 0 points. then we might have to create appropriate starting points that are not in the index set. Please see ::set_start_point_at_random

Reply:
we may not support this for streaming settings?